File: import.py
#!/usr/bin/env python

import glob
import os
import sys
import shutil

from lib.database_wrapper import DatabaseWrapper
from lib.resize import resize

from settings import config

db = DatabaseWrapper(config.CONNECTION_STRING)

counter = 0
for filename in glob.glob(config.SOURCE_PATH + '*'):
    if os.path.isfile(filename):
        counter += 1

        basename = os.path.basename(filename)

        dest_name = config.DESTINATION_ROOT + 'images/' + basename
        if os.path.exists(dest_name):
            print('{}: File {} already found'.format(counter, basename))
        else:
            print('{}: Need to import {}'.format(counter, basename))

            (path, ext) = os.path.splitext(basename)
            othername = path + '.png'

            db.add_new_photo(basename, othername)

            # Create the medium image
            new_filename = config.DESTINATION_ROOT + 'medium/' + othername
            resize(filename, new_filename, 800)

            # Create the thumbnail
            new_filename = config.DESTINATION_ROOT + 'thumbs/' + othername
            resize(filename, new_filename, 125)

            # Copy the source image
            new_filename = config.DESTINATION_ROOT + 'images/' + basename
            shutil.copy2(filename, new_filename)

        # Remove the original
        os.remove(filename)

File: import.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'yaml'
require 'sequel'
require 'fileutils'

$LOAD_PATH << './lib'

require 'database_access'
require 'resize'

config = YAML.load_file('config.yaml')

da = DatabaseAccess.new(config['dcs'])

counter = 0

now_is = Time.now

Dir["#{config['source_path']}*"].each do |filename|
  counter += 1

  basename = File.basename(filename)

  dest_name = "#{config['destination_root']}images/#{basename}"

  if File.exist?(dest_name)
    puts("#{counter}: File #{basename} already found")
  else
    puts("#{counter}: Need to import #{basename}")

    othername = basename.split('.')[0..-2].join('.') + '.png'

    file_size = File.size(filename)
    photo_id = da.add_new_photo(basename, othername, file_size, now_is)
    da.add_tag_to_photo(photo_id, 'untagged')

    # Create the medium image
    new_filename = "#{config['destination_root']}medium/#{othername}"
    resize(filename, new_filename, 800)

    # Create the thumbnail
    new_filename = "#{config['destination_root']}thumbs/#{othername}"
    resize(filename, new_filename, 125)

    # Copy the source image
    new_filename = "#{config['destination_root']}images/#{basename}"
    FileUtils.cp(filename, new_filename)
  end

  File.delete(filename)
end
